server/resources.py

```diff
@@ -61,5 +62,43 @@ def post(self, lecture_id):
             'id': comment.id
         }
 
+
+class EngagementListResource(Resource):
+    def post(self, lecture_id):
+        lecture = Lecture.query.filter(Lecture.id == lecture_id).first()
+
+        if not lecture:
+            abort(404, message="Lecture {} does not exist".format(lecture_id))
+
+        parser = reqparse.RequestParser()
+        parser.add_argument('challenge', required=True, type=float)
+        parser.add_argument('interest', required=True, type=float)
+        parser.add_argument('time', required=True, type=str)
+        args = parser.parse_args()
+
+        if (args.challenge < 0.0 or args.challenge > 1.0):
+            abort(400, message="Challenge must be in range [0,1]")
+
+        if (args.interest < 0.0 or args.interest > 1.0):
+            abort(400, message="Interest must be in range [0,1]")
+
+        challenge = args.challenge
```

Comment:
Hm, I think I just thought like checking if the arguments are valid first, then actually storing the argument. But it's probably better to do it before the if so that args doesn't have to be dereferenced each time.